admin/code/LeftAndMain.php

```diff
@@ -1260,14 +1353,27 @@ public function getEditForm($id = null, $fields = null) {
 			$actionsFlattened = $actions->dataFields();
 			if($actionsFlattened) foreach($actionsFlattened as $action) $action->setUseButtonTag(true);
 
-			$form = CMSForm::create(
+			$negotiator = $this->getResponseNegotiator();
+			$form = Form::create(
 				$this, "EditForm", $fields, $actions
 			)->setHTMLID('Form_EditForm');
-			$form->setResponseNegotiator($this->getResponseNegotiator());
 			$form->addExtraClass('cms-edit-form');
 			$form->loadDataFrom($record);
 			$form->setTemplate($this->getTemplatesWithSuffix('_EditForm'));
 			$form->setAttribute('data-pjax-fragment', 'CurrentForm');
+			$form->setValidationResponseCallback(function() use ($negotiator, $form) {
```

Comment:
Won't you need to set this callback on the other leftandmain subclasses too? E.g. SecurityAdmin swaps out CMSForm for Form, but doesn't have this code. Also ModelAdmin.